### PR TITLE
[Related #42] Develop home(join/create room) page

### PR DESCRIPTION
**Pull request by supunTE**
-  Develop home(join/create room) page
- Initialized i18n (which helps us to maintain text content separate from the UI, also we can use it to translate in the future

### CODE REVIEW

File: apps/client/src/components/molecules/button/Button.component.tsx [hunk 1]
```diff
@@ -48,7 +50,8 @@ export function Button({
                   isButtonDanger,
               },
               { 'bg-transparent hover:bg-neutral-20': !isButtonDanger },
-              'border hover:border-4 border-neutral-40',
+              'border hover:ring-[4px] border-neutral-40 ring-neutral-40',
+              // 'm-[4px] hover:m-[1px]',
```
Comment (dalanad) [inline apps/client/src/components/molecules/button/Button.component.tsx]: commented code

File: apps/client/src/components/atoms/text/Text.component.tsx [hunk 2]
```diff
@@ -66,24 +73,49 @@ export function Text({
...
 
   /** Related text size to the text variant */
   const textSizeValue = TextStyles.TEXT_MAP.get(textVariant);
-  const classNames = (colorClass || 'text-neutral-90') + ' ' + className;
+  const classNames =
+    (colorClass || 'text-neutral-90') +
+    ' ' +
+    className +
+    ' ' +
+    (isAnimatable && 'transition-all duration-300');
```
Comment (dalanad) [inline apps/client/src/components/atoms/text/Text.component.tsx]: image

Comment (dalanad) [reply apps/client/src/components/atoms/text/Text.component.tsx]: Use the ternary operator

**Review by dalanad — CHANGES_REQUESTED** [review]

File: apps/client/src/components/atoms/text/Text.component.tsx (hunk 2 above)
Comment (supunTE) [reply apps/client/src/components/atoms/text/Text.component.tsx]: fixed 7b3db10bf048eb14a05ac33cbb729fc19f70516e

File: apps/client/src/components/molecules/button/Button.component.tsx (hunk 1 above)
Comment (supunTE) [reply apps/client/src/components/molecules/button/Button.component.tsx]: fixed 9ce158d9a86ba595892ad0b06d61a1f4f8a45811

File: apps/client/src/components/atoms/text/Text.component.tsx [hunk 3]
```diff
@@ -66,24 +73,49 @@ export function Text({
...
+          {content}
         </p>
       );
     }
   };
 
   /** Related text size to the text variant */
   const textSizeValue = TextStyles.TEXT_MAP.get(textVariant);
-  const classNames = (colorClass || 'text-neutral-90') + ' ' + className;
+  const classNames =
```
Comment (pasinducw) [inline apps/client/src/components/atoms/text/Text.component.tsx:98]: Why aren't you using `classnames` lib to build the classnames here?

File: package.json [hunk 4]
```diff
@@ -33,11 +33,14 @@
...
     "dotenv": "^16.0.3",
     "express": "^4.18.2",
+    "i18next": "^22.4.9",
     "lodash": "^4.17.21",
     "nanoid": "^3.3.4",
     "pixelarticons": "^1.7.0",
     "react": "18.2.0",
     "react-dom": "18.2.0",
+    "react-i18next": "^12.1.5",
+    "react-linkify": "^1.0.0-alpha",
```
Comment (pasinducw) [inline package.json]: Why do we need this library?

**Review by pasinducw — CHANGES_REQUESTED** [review]

File: apps/client/src/components/atoms/text/Text.component.tsx (hunk 3 above)
Comment (supunTE) [reply apps/client/src/components/atoms/text/Text.component.tsx:98]: already fixed in 9953c3957419625c23d0ca0cccf993ac3ea6eea9 under PR #51

File: package.json (hunk 4 above)
Comment (supunTE) [reply package.json]: - react-i18next
0f9a9f9511cd39d8b5449ba15540b4d628690c4f
add UI strings to i18n to JSON file. so we don't have to handle text inside layout files. also,
we can use them to translate websites in the future.

- linkfy
this package is using for parse links on the text
image
inside text component children will passed into this package. if there any links inside of the text they will wrap in `<a></a>` tags

Comment (supunTE) [reply package.json]: Removed react-linkify package (2c257c4b6d75666525cfd58fa0f67d0f5f6fdae9) and create link subcomponent to render links (Also used i18next interpolation to parse link) (2bbf2933205ac1de623519507b546aa0a7e3402e)

**Review by dalanad — APPROVED** [review]

**Review by pasinducw — APPROVED** [review]

**Pull request merged**